apollo-router/src/plugins/telemetry/metrics/apollo.rs

```diff
@@ -0,0 +1,923 @@
+//! Apollo metrics
+// This entire file is license key functionality
+use crate::plugins::telemetry::apollo::Config;
+use crate::plugins::telemetry::config::MetricsCommon;
+use crate::plugins::telemetry::metrics::{MetricsBuilder, MetricsConfigurator};
+use crate::stream::StreamExt;
+use apollo_spaceport::{
+    ContextualizedStats, ReferencedFieldsForType, Reporter, ReporterError, ReporterGraph,
+    StatsContext,
+};
+use async_trait::async_trait;
+use deadpool::{managed, Runtime};
+use futures::channel::mpsc;
+use futures_batch::ChunksTimeoutStreamExt;
+use serde::Serialize;
+use std::collections::HashMap;
+use std::ops::AddAssign;
+use std::time::Duration;
+use tower::BoxError;
+use url::Url;
+
+const DEFAULT_BATCH_SIZE: usize = 65_536;
+const DEFAULT_QUEUE_SIZE: usize = 65_536;
+
+#[derive(Clone)]
+pub(crate) enum Sender {
+    Noop,
+    Spaceport(mpsc::Sender<Metrics>),
+}
+
+impl Sender {
+    pub(crate) fn send(&self, metrics: Metrics) {
+        match &self {
+            Sender::Noop => {}
+            Sender::Spaceport(channel) => {
+                if let Err(err) = channel.to_owned().try_send(metrics) {
+                    tracing::warn!(
+                        "could not send metrics to spaceport, metric will be dropped: {}",
+                        err
+                    );
+                }
+            }
+        }
+    }
+}
+
+impl Default for Sender {
+    fn default() -> Self {
+        Sender::Noop
+    }
+}
+
+#[derive(Default, Debug, Serialize)]
+pub(crate) struct Metrics {
+    pub(crate) client_name: String,
+    pub(crate) client_version: String,
+    pub(crate) stats_report_key: String,
+    pub(crate) query_latency_stats: QueryLatencyStats,
+    pub(crate) per_type_stat: HashMap<String, TypeStat>,
+    pub(crate) referenced_fields_by_type: HashMap<String, ReferencedFieldsForType>,
+    pub(crate) operation_count: u64,
+}
+
+#[derive(Default, Debug, Serialize)]
+pub(crate) struct QueryLatencyStats {
```

Comment:
`latency_count` mirrors the protobuf format, can you elaborate on why this should be converted to bools?

Reply:
What's confusing is that you've changed the type of latency_count from DurationHistogram (a thing that can represent multiple requests) to Duration (a thing that can only represent a single request) but you've left most of the other fields which count requests as numbers. I don't think there's any case you would ever set those numbers to something greater than 1 because any such case would also require you to have more than one "duration".